docs/CIP_5_Large_Batch_Handling_Improvements.md

```diff
@@ -0,0 +1,56 @@
+# CIP-5: Large Batch Handling Improvements Proposal
+
+## Status
+
+Current Status: `Under Discussion`
+
+## **Motivation**
+
+As users start putting Chroma in its paces and storing ever-increasing datasets, we must ensure that errors
+related to significant and potentially expensive batches are handled gracefully. This CIP proposes to add a new
+setting, `max_batch_size` API, on the local segment API and use it to split large batches into smaller ones.
+
+## **Public Interfaces**
+
+The following interfaces are impacted:
+
+- New Server API endpoint - `/pre-flight-checks`
+- New `max_batch_size` property on the `API` interface
+- Updated `_add` method on `chromadb.api.segment.SegmentAPI`
+- Updated `_add` method on `chromadb.api.fastapi.FastAPI`
+
+## **Proposed Changes**
+
+We propose the following changes:
+
+- The new `max_batch_size` property is now available in the `API` interface. The property relies on the
+  underlying `Producer` class
+  to fetch the actual value. The property will be implemented by both `chromadb.api.segment.SegmentAPI`
+  and `chromadb.api.fastapi.FastAPI`
+- `chromadb.api.segment.SegmentAPI` will implement the `max_batch_size` property by fetching the value from the
+  `Producer` class.
+- `chromadb.api.fastapi.FastAPI` will implement the `max_batch_size` by fetching it from a new `/pre-flight-checks`
```

Comment:
is the assumption that these are fetched once on client creation?

Reply:
Haven't thought about it but actual impl caches the values

Reply:
https://github.com/amikos-tech/chroma-core/blob/93f0eda92ff1061a42cb6b3cf219ce6291c6f4ef/chromadb/api/fastapi.py#L400